### ISSUE TITLE
Internal test script failed

### ISSUE THREAD

**Issue by laffernandes** [joaofazolo/boca-docker#35]
Sorry to bother you again. I created users and uploaded an example problem (abacaxi.zip) successfully. However, when submitting a solution, the following error appears on the screen showing the run details: `error: internal test script failed (/home/bocajail/tmp/bocaVuBz4N.d/problemdata/tests/c`. It seems to be a problem compiling the script that tests whether autojudge is working. What could I be doing wrong?

**Comment by laffernandes** [comment]
Another problem (maybe related) is that the `.zip` is not being created using the `http://localhost:8000/boca/admin/buildproblem.php` tool. The error is that the `[...]/doc/problemexamples/[...]` folder is missing in the `boca-web` container. I fixed this problem by copying the `https://github.com/cassiopc/boca/tree/master/doc/problemexamples` folder to `var/www/boca/doc/problemexamples/[...]` in the `boca-web` container. Now, the `.zip` is being created without the `test` scripts (at least it is being created). Even when putting the test scripts manually, the error mentioned above occurs.

**Comment by blegora** [comment]
> Sorry to bother you again. I created users and uploaded an example problem (abacaxi.zip) successfully. However, when submitting a solution, the following error appears on the screen showing the run details: `error: internal test script failed (/home/bocajail/tmp/bocaVuBz4N.d/problemdata/tests/c`. It seems to be a problem compiling the script that tests whether autojudge is working. What could I be doing wrong?

I think this is a problem involving Boca's 'safeexec' script and gcc and/or docker (I only tested it using boca-docker, so I don't know if it only happens there). Initially I had added a custom zip to BOCA's problems, with an empty "tests" folder, and it was returning "resource unavailable" errors. I discovered a workaround to solve my issue after many attemps, and I think this is also related to your problem.

I discovered that by increasing safeexec's child process limit, these errors stop occurring and autojudge works. To do this, I had to edit the compile/c file, adding "-u1024" to the lines that call safeexec (I tested with smaller values, like 256 and 512, but without success). 

Can you test if the following file works? [abacaxi_edited.zip](https://github.com/joaofazolo/boca-docker/files/13325450/abacaxi_edited.zip)


> Another problem (maybe related) is that the `.zip` is not being created using the `http://localhost:8000/boca/admin/buildproblem.php` tool. The error is that the `[...]/doc/problemexamples/[...]` folder is missing in the `boca-web` container. I fixed this problem by copying the `https://github.com/cassiopc/boca/tree/master/doc/problemexamples` folder to `var/www/boca/doc/problemexamples/[...]` in the `boca-web` container. Now, the `.zip` is being created without the `test` scripts (at least it is being created). Even when putting the test scripts manually, the error mentioned above occurs.

Same problem here... Followed your steps and successfully created a .zip. However, if i add the created zip as a problem, and submit a solution, autojudge returns "error: problem package file is invalid (6)". And I think that even if it was valid,, we had to edit compile/c to solve the first issue...

**Comment by rlaiola** [comment]
> Sorry to bother you again. I created users and uploaded an example problem (abacaxi.zip) successfully. However, when submitting a solution, the following error appears on the screen showing the run details: `error: internal test script failed (/home/bocajail/tmp/bocaVuBz4N.d/problemdata/tests/c`. It seems to be a problem compiling the script that tests whether autojudge is working. What could I be doing wrong?

@laffernandes I finally managed to take a look at the reported issue but could not reproduce it (works fine here). In case it is still relevant to you, try the following:
- Start and setup BOCA as indicated [here](https://github.com/joaofazolo/boca-docker?tab=readme-ov-file#quick-start);
- Get the `abacaxi` problem from [https://github.com/cassiopc/boca/tree/master/doc/problemexamples/abacaxi](https://github.com/cassiopc/boca/tree/master/doc/problemexamples/abacaxi);
- Compress it accordingly: `zip -r abacaxi.zip . # make sure it is the working directory`;
- Submit a solution as a team and check the output as a admin user;

codigo.c (wrong solution):
```c
#include <math.h>
#include <stdio.h>

int main(){

	int x,y;
	scanf ("%d %d", &x, &y);
	float soma = sqrt(x) + sqrt(y);

	printf ("%.2f", soma);

	return 0;
	
}
```

Below the output:
![Screen Shot 2024-02-22 at 12 47 10](https://github.com/joaofazolo/boca-docker/assets/4059310/65c193da-3316-4153-b1e2-e04f515a6099)

Stdout:
![Screen Shot 2024-02-22 at 12 52 42](https://github.com/joaofazolo/boca-docker/assets/4059310/e34f91f2-8f36-4f27-b65f-b309c55d6a3d)

Stderr:
![Screen Shot 2024-02-22 at 12 52 36](https://github.com/joaofazolo/boca-docker/assets/4059310/a8ed620d-d8a1-4fec-97b6-8824e008f819)

**Comment by rlaiola** [comment]
> Another problem (maybe related) is that the `.zip` is not being created using the `http://localhost:8000/boca/admin/buildproblem.php` tool. The error is that the `[...]/doc/problemexamples/[...]` folder is missing in the `boca-web` container. I fixed this problem by copying the `https://github.com/cassiopc/boca/tree/master/doc/problemexamples` folder to `var/www/boca/doc/problemexamples/[...]` in the `boca-web` container. Now, the `.zip` is being created without the `test` scripts (at least it is being created). Even when putting the test scripts manually, the error mentioned above occurs.

@laffernandes Indeed this was a bug. Fixed in PR #47. The nightly build images will include the fix by tomorrow (12:00 PM UTC) and the latest tag ones by this Monday (around same time).

Cheers!

**Comment by rlaiola** [comment]
@laffernandes @blegora

If possible, please verify if the latest images (tags `latest`, `1.2` or `1.2.2`) solves the problem with safeexec.

Cheers!
